src/reflect.ts

```diff
@@ -20,3 +20,4 @@ export type Reflect =
   | { tag: 'intersect'; intersectees: Reflect[] } & Runtype<always>
   | { tag: 'function' } & Runtype<(...args: any[]) => any>
   | { tag: 'constraint'; underlying: Reflect; constraint: ConstraintCheck<Reflect>; args?: any } & Runtype<always>
+  | { tag: 'instanceof', ctor: Reflect } & Runtype<any>
```

Comment:
Should be `ctor: Function` probably? The underlying constructor is not a runtype, so it won't have a `Reflect`.